src/Platforms/SYCL/syclBLAS.cpp

```diff
@@ -182,6 +182,109 @@ template sycl::event gemm(sycl::queue& handle,
                           const int ldc,
                           const std::vector<sycl::event>& events);
 
+//transpose
+template<typename T1, typename T2>
+sycl::event transpose(sycl::queue& q,
+                      const T1* restrict in,
+                      int m,
+                      int lda,
+                      T2* restrict out,
+                      int n,
+                      int ldb,
+                      const std::vector<sycl::event>& events)
+{
+  constexpr size_t tile_size = 16;
+  const size_t m_max         = ((m + tile_size - 1) / tile_size) * tile_size;
+  const size_t n_max         = ((n + tile_size - 1) / tile_size) * tile_size;
+
+  return q.submit([&](sycl::handler& cgh) {
+    cgh.depends_on(events);
+    sycl::accessor<T2, 2, sycl::access::mode::write, sycl::access::target::local> tile(sycl::range<2>(tile_size,
+                                                                                                      tile_size + 1),
+                                                                                       cgh);
+
+    cgh.parallel_for(sycl::nd_range<2>{{m_max, n_max}, {tile_size, tile_size}}, [=](sycl::nd_item<2> item) {
```

Comment:
I see. Accessor needs cgh.

Q: the command group submit to q can be executed asynchronously. Should m_max, n_max captured by value instead of reference? tile_size is constexpr, should not be an issue. Should in, m, lda, out, n, ldb all capture by value?

Reply:
A: the host-side part of the command group will finish before the call to submit exits 
https://developer.codeplay.com/products/computecpp/ce/2.8.0/guides/sycl-guide/hello-sycl
So the & capture at q.submit is OK.